app/src/main/java/com/randomappsinc/studentpicker/home/MainActivity.java

```diff
@@ -310,4 +323,42 @@ public boolean onOptionsItemSelected(MenuItem item) {
         }
         return super.onOptionsItemSelected(item);
     }
+
+    private void showRenameDialog(final int position) {
+        new MaterialDialog.Builder(this)
```

Comment:
364 lines is pretty large. Let's create a new class in the home package called "RenameListDialog". It will communicate with the activity through an interface (the method can be something like onRenameListConfirmed(int position, String newName). The activity will invoke the dialog through a method like show(int position). This new class will be responsible for creating and showing the dialog, allowing the activity to have less code. We also don't need to keep creating the same dialog multiple times. We can store a reference to the dialog in this new class and clear out the input whenever we need to show it again.